zilliqa-macros/src/test.rs

```diff
@@ -27,7 +27,7 @@ pub(crate) fn test_macro(_args: TokenStream, item: TokenStream) -> TokenStream {
                 vec![seed.to_str().unwrap().parse().unwrap()]
             } else {
                 let samples: usize = std::env::var_os("ZQ_TEST_SAMPLES")
-                    .map(|s| s.to_str().unwrap().parse().unwrap())
+                    .map(|s| s.to_str().unwrap_or("1").parse().unwrap_or(1))
```

Comment:
this threw an error if it couldn't parse the string (trailing whitespace possibly), but you can't tell where in the macro it came from

Reply:
A bit odd (trailing whitespace should never be part of an env variable in most shells... I would assume at least), also the same issue would probably happen with `ZQ_TEST_RNG_SEED`.

What context were you encountering the error in? I'd argue that it might be better to fail early, especially e.g. in CI we definitely don't want a parse error to be silently converted to 1 and thus believe we're running lots of tests when actually defaulting to a single sample.

Reply:
Yes, that is fair. The reason that I defaulted it to 1 was because the original code is:

```
                let samples: usize = std::env::var_os("ZQ_TEST_SAMPLES")
                    .map(|s| s.to_str().unwrap().parse().unwrap())
                    .unwrap_or(1);
```

And I just followed its intent - to default to 1. But I do agree, it should panic in a legible way if it cannot read it, I will change it.

The context is that my IDE (misconfigured by me) set this to a non-integer.

Reply:
Yeah, the original unwraps the option (i.e. if the variable is not specified at all), but panics on the error.